### PR TITLE
ARTEMIS-743 Default the queue address to the queue name

### PR DESCRIPTION
**Pull request by franz1981**
https://issues.apache.org/jira/browse/ARTEMIS-743

### CODE REVIEW

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/QueueConfig.java [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+
+import org.apache.activemq.artemis.api.core.SimpleString;
+import org.apache.activemq.artemis.core.filter.Filter;
+import org.apache.activemq.artemis.core.filter.FilterUtils;
+import org.apache.activemq.artemis.core.paging.PagingManager;
+import org.apache.activemq.artemis.core.paging.cursor.PageSubscription;
+
+public final class QueueConfig {
+
+   private final long id;
```
Comment (clebertsuconic) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/server/QueueConfig.java]: The id is assigned when the queue is persisted. Why this is part of the config options?

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ActiveMQServerImpl.java [hunk 2]
```diff
@@ -1290,10 +1291,7 @@ protected ServerSessionImpl internalCreateSession(String name,
...
                                                      boolean autoCreateJMSQueues) throws Exception {
-      return new ServerSessionImpl(name, username, password, validatedUser, minLargeMessageSize, autoCommitSends, autoCommitAcks, preAcknowledge, configuration.isPersistDeliveryCountBeforeDelivery(),
-                                   xa, connection, storageManager, postOffice, resourceManager, securityStore, managementService, this, configuration.getManagementAddress(),
-                                   defaultAddress == null ? null : new SimpleString(defaultAddress), callback, context, autoCreateJMSQueues ? jmsQueueCreator : null,
-                                   pagingManager);
+      return new ServerSessionImpl(name, username, password, validatedUser, minLargeMessageSize, autoCommitSends, autoCommitAcks, preAcknowledge, configuration.isPersistDeliveryCountBeforeDelivery(), xa, connection, storageManager, postOffice, resourceManager, securityStore, managementService, this, configuration.getManagementAddress(), defaultAddress == null ? null : new SimpleString(defaultAddress), callback, context, autoCreateJMSQueues ? jmsQueueCreator : null, pagingManager);
```
Comment (clebertsuconic) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ActiveMQServerImpl.java]: this line gets too big.. can you break it?

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/QueueConfig.java (hunk 1 above)
Comment (clebertsuconic) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/server/QueueConfig.java]: ahhh... this is an internal class. I thought it was part of the interface now.

Never mind then

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ActiveMQServerImpl.java (hunk 2 above)
Comment (franz1981) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ActiveMQServerImpl.java]: Yes! 
And then i'll squash the commit!

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/QueueConfig.java [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+      private boolean durable;
+      private boolean temporary;
+      private boolean autoCreated;
+
+      private Builder(final long id, final SimpleString name) {
+         this(id, name, name);
+      }
+
+      private Builder(final long id, final SimpleString name, final SimpleString address) {
+         //TODO Checks if what are valid default values and/or enforces not mandatory fields to be on Builder's constructor
```
Comment (clebertsuconic) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/server/QueueConfig.java]: I would remove this TODO

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/QueueConfig.java [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+         this.filter = null;
+         this.pagingManager = null;
+         this.user = null;
+         this.durable = true;
+         this.temporary = false;
+         this.autoCreated = true;
+         validateState();
+      }
+
+      //TODO maybe an SimpleString::isEmpty method would be welcome
```
Comment (clebertsuconic) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/server/QueueConfig.java]: I would remove this TODO as well.

**Comment by clebertsuconic** [comment]
I am happy with this. +1 for merge...

Two small TODOS I would remove now.. but other than that good to merge...

Will let @mtaylor also to comment on this.

**Comment by clebertsuconic** [comment]
just to let you guys know.. I ran the testsuite and only got a few known failures. looks good.

**Pull request closed without merging**